packages/firebase_storage/firebase_storage/example/lib/save_as/save_as_html.dart

```diff
@@ -2,48 +2,61 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-// ignore: avoid_web_libraries_in_flutter
-import 'dart:html';
+import 'dart:js_interop';
 import 'dart:typed_data';
 
+import 'package:web/web.dart' as web;
+
 /// Initializes a DOM container where we can host elements.
-Element _ensureInitialized(String id) {
-  var target = querySelector('#$id');
+web.Element _ensureInitialized(String id) {
+  var target = web.document.querySelector('#$id');
   if (target == null) {
-    final Element targetElement = Element.tag('flt-x-file')..id = id;
+    final web.Element targetElement =
+        web.document.createElement('flt-x-file') as web.HTMLElement;
+    targetElement.id = id;
 
-    querySelector('body')?.children.add(targetElement);
+    web.document.querySelector('body')?.children.add(targetElement);
     target = targetElement;
   }
   return target;
 }
 
-AnchorElement _createAnchorElement(String href, String suggestedName) {
-  return AnchorElement(href: href)..download = suggestedName;
+web.HTMLAnchorElement _createAnchorElement(String href, String suggestedName) {
+  final element = web.HTMLAnchorElement();
+  element.href = href;
+  element.download = suggestedName;
+  return element;
 }
 
 /// Add an element to a container and click it
-void _addElementToContainerAndClick(Element container, Element element) {
+void _addElementToContainerAndClick(
+  web.Element container,
+  web.Element element,
+) {
   // Add the element and click it
   // All previous elements will be removed before adding the new one
   container.children.add(element);
-  element.click();
+  final event = web.MouseEvent('click');
+  element.dispatchEvent(event);
 }
 
 /// Present a dialog so the user can save as... a bunch of bytes.
 Future<void> saveAsBytes(Uint8List bytes, String suggestedName) async {
   // Convert bytes to an ObjectUrl through Blob
-  final blob = Blob([bytes]);
-  final path = Url.createObjectUrl(blob);
+  final blob = web.Blob([bytes.toJS].toJS);
+  final path = web.URL.createObjectURL(blob);
 
   // Create a DOM container where we can host the anchor.
   final target = _ensureInitialized('__x_file_dom_element');
 
   // Create an <a> tag with the appropriate download attributes and click it
   // May be overridden with XFileTestOverrides
-  final AnchorElement element = _createAnchorElement(path, suggestedName);
+  final web.HTMLAnchorElement element =
+      _createAnchorElement(path, suggestedName);
 
   // Clear the children in our container so we can add an element to click
-  target.children.clear();
+  for (int i = 0; i < target.children.length; i++) {
```

Comment:
I don't think this will work. The length will reduce every call, so maybe only n/2 items will be removed.

Maybe better to do a `do/while`?

Reply:
I think we only have a length of 1, but maybe it's isn't the case in some places. 
Updated the example app code here: https://github.com/firebase/flutterfire/pull/12614